covscan: fix miscellaneaous errors

jeromemarchand opened the pull request:
This fixes a few errors discovered by a coverity scan.

jeromemarchand commented on src/cc/bcc_elf.c:
```diff
@@ -373,6 +373,9 @@ static int verify_checksum(const char *file, unsigned int crc) {
   void *buf;
   unsigned int actual;
 
+  if (file == NULL)
```
Good point! Commit b84714a47a3a1ec probably made the coverity warning disappeared already.

jeromemarchand commented on src/cc/bcc_proc.c:
```diff
@@ -99,7 +99,7 @@ int bcc_procutils_each_module(int pid, bcc_procutils_modulecb callback,
     buf[0] = '\0';
     // From fs/proc/task_mmu.c:show_map_vma
-    if (fscanf(procmap, "%lx-%lx %s %llx %s %lu%[^\n]", &begin, &end, perm,
+    if (fscanf(procmap, "%lx-%lx %4s %llx %7s %lu%[^\n]", &begin, &end, perm,
```
It's 7 because the buffer dev is of size 8. But a 6 bytes buffer would be enough. We don't do anything with it after the call to fscanf() anyway.

yonghong-song replied:
Then "7" is okay. thanks for explanation.

palmtenor replied:
Just want to be sure, `%Ns` means reading **at most** `N` characters, right? Dev number could definitely be shorter.

palmtenor commented on src/cc/bcc_elf.c:
```diff
@@ -435,8 +438,10 @@ static char *find_debug_via_debuglink(Elf *e, const char *binpath,
 DONE:
   free(bindir);
-  if (res && check_crc && !verify_checksum(res, crc))
+  if (res && check_crc && !verify_checksum(res, crc)) {
```
Huh, I just looked at Man page for `dirname` and it says
```
These functions may return pointers to statically allocated memory which may be overwritten by subsequent calls. Alternatively, they may return a pointer to some part of path, so that the string referred to by path should not be modified or freed until the pointer returned by the function is no longer required.
```
So maybe the `free(bindir)` part is also not really correct per say?

yonghong-song replied:
I just checked glibc code, For the path without '/' (e.g., a.out), a static memory inside dirname is returned.
It looks the best way is to declare an array, e.g., `bindir[PATH_MAX]` as the input to the bindir function.

This code will not execute in typical case. It will be executed if a separate debug file is specified, which is probably we won't hit the issue often.

jeromemarchand replied:
Good point! The result of strdup(binpath) should be preserved in a tmp pointer and that pointer freed instead of bindir.

jeromemarchand commented:
Did that last update addressed all of your concerns? I dropped the unneeded patch, reduced the dev buffer to 6 bytes and fixed the free()/dirname() issue.

yonghong-song commented:
Sorry, I did not notice you made the change. The change actually looks good. Will test it now.

yonghong-song commented:
[buildbot, test this please]

yonghong-song commented:
[buildbot, test this please]

yonghong-song commented:
[buildbot, test this please]
